### PR TITLE
Added ls7366r driver for ESP-IDF

### PR DESCRIPTION
**Pull request by Jkallus**
I've written a driver for the LS7366R Quadrature Encoder Counter. This is my first time contributing code so please let me know if there are any steps I've missed.

 Thanks Josh

### CODE REVIEW

File: components/ls7366r/ls7366r.h [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	* @param enable counter enabled or disabled
+	* @return `ESP_OK` on success
+	*/
+	esp_err_t ls7366r_counter_enable(ls7366r_t *dev, bool enable);
+	
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* __LS7366R_H__ */
```
Comment (UncleRus) [inline components/ls7366r/ls7366r.h:226]: newline at the end of the file is required

**Comment by UncleRus** [comment]
Thank you!
To merge this PR, please do the following: 

1. Please add an example of using the component.
2. You also need to add the component in the `README.md` to the "Input controls" table. Don't forget to add your information to the contributors list at the end of the file.
3. You also need to create a section in the documentation. For this:
- create a file `docs/source/groups/ls7366r.rst` with content similar to other files in this directory
- edit `docs/source/index.rst`, adding a link to the created documentation section (`   groups/ls7366r`) to the Input controls table of contents (insert line after line 210)

PS CI build was failed because of the bug in CI script that has already been fixed.

**Comment by Jkallus** [comment]
Okay sounds good. It may be a few days till I can do it but that all sounds easy enough.

**Comment by UncleRus** [comment]
@Jkallus Any news?

**Comment by Jkallus** [comment]
Hi, sorry for the delay. I haven't gotten around to it yet because I ended up not needing the decoder for what I'm making. I still would like to finish it. Thank you for the reminder I intend to do it during this week.

**Comment by Jkallus** [comment]
Okay it should be finished now, sorry for the delay

**Comment by trombik** [comment]
@Jkallus would you add a metadata file to the component?

https://github.com/UncleRus/esp-idf-lib/blob/master/Metadata.md

for tl;dr, see metadata files in other components.

after adding the metadata file, you need to update `README.md`. see https://github.com/UncleRus/esp-idf-lib/blob/master/Metadata.md#usages-of-metadata-in-the-project 

and 

https://github.com/UncleRus/esp-idf-lib/blob/master/Metadata.md#validating-metadata-of-components

**Comment by trombik** [comment]
@Jkallus also, your example failed to build. see https://github.com/UncleRus/esp-idf-lib/runs/4404016215?check_suite_focus=true

**Pull request merged**